apollo-router/src/http_server_factory.rs

```diff
@@ -67,7 +62,15 @@ impl HttpServerHandle {
         if let Err(_err) = self.shutdown_sender.send(()) {
             tracing::error!("Failed to notify http thread of shutdown")
         };
-        self.server_future.await.map(|_| ())
+        #[allow(unused_variables)]
+        let listener = self.server_future.await?;
```

Comment:
Small NIT: I prefer to use let _, rather than this particular clippy directive.

Reply:
Oh actually that's a mistake! The previous code was using the listener to get the socket addr but this version uses the stored one (slightly easier).

```suggestion
        let _listener = self.server_future.await?;
```